tracker.js

```diff
@@ -35,6 +36,33 @@ function track() {
         if (vcapApplication.application_uris) {
             event.application_uris = vcapApplication.application_uris;
         }
+        if (process.env.VCAP_SERVICES) {
+            // refer to http://docs.cloudfoundry.org/devguide/deploy-apps/environment-variable.html#VCAP-SERVICES
+            vcapServices =  JSON.parse(process.env.VCAP_SERVICES);
+            if(Object.keys(vcapServices).length > 0) {
+                event.bound_vcap_services = {};
```

Comment:
If `process.env.VCAP_APPLICATION` is _not_ defined but `process.env.VCAP_SERVICES` _is_ defined, then `event` will not be defined at this point, resulting in a `ReferenceError`. To add a bit of defensive coding, we should initialize `event` as an empty object (`{}`) before the `if` statement on line 17.